sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java

```diff
@@ -294,7 +295,7 @@ private static PodSpec convertPod(
 
         Collection<URI> podUris = new ArrayList<>();
         for (String uriStr : rawPod.getUris()) {
-            podUris.add(new URI(uriStr));
+            podUris.addAll(getUris(uriStr));
```

Comment:
If we pass the return of `rawPod.getUris()` to `getUris`, we could use a `flatMap` to handle them all as a single list:
```java
        Collection<String> allUris = rawPod.getUris().stream()
                .flatMap(uris -> Arrays.stream(uris.split(",")))
                .map(u -> u.trim())
                .filter(StringUtils::isNotBlank)
                .collect(Collectors.toList());
```

Reply:
Tried. Streaming method doesn't handle exceptions well.

Reply:
Yes, I also saw that w.r.t the URI constructor, but in this case, we're just getting a list of all non-blank URI strings. But this is definitely out of scope here, and was just me thinking aloud.